scripts/backup-restore.sh

```diff
@@ -0,0 +1,401 @@
+    if [ -z "$snapshot_repo" ]; then
+      log "ERROR: Could not find OpenSearch snapshot repository."
+      exit 1
+    fi
+    log "Found snapshot repository: ${snapshot_repo}"
+
+    log "Finding latest successful snapshot..."
+    local latest_snapshot
+    latest_snapshot=$(curl -kL -s -u "${user}:${password}" -X GET "${os_url}/_snapshot/${snapshot_repo}/_all" | jq -r '.snapshots | map(select(.state == "SUCCESS")) | sort_by(.start_time_in_millis) | .[-1].snapshot')
+    if [ -z "$latest_snapshot" ] || [ "$latest_snapshot" = "null" ]; then
+      log "ERROR: No successful snapshots found to restore."
+      exit 1
+    fi
+    log "Found latest snapshot to restore: ${latest_snapshot}"
+
+    local indices_to_restore="kubernetes-*,kubeaudit-*,other-*,authlog-*"
+    log "Will restore indices matching pattern: ${indices_to_restore}"
+
+    log "Closing existing indices matching the restore pattern..."
+    curl -kL -s -u "${user}:${password}" -X POST "${os_url}/${indices_to_restore}/_close?pretty"
+
+    log "Starting restore from snapshot ${latest_snapshot}..."
+    curl -kL -s -u "${user}:${password}" -X POST "${os_url}/_snapshot/${snapshot_repo}/${latest_snapshot}/_restore?pretty" -H 'Content-Type: application/json' -d'
+      {
+        "indices": "'"${indices_to_restore}"'"
+      }
+      '
+
+    log "Waiting for cluster health to become green..."
+    local status=""
+    local retries=60 # 30 min timeout
+    while [ $retries -gt 0 ]; do
+      status=$(curl -kL -s -u "${user}:${password}" "${os_url}/_cluster/health" | jq -r '.status')
+      log "Current cluster status: ${status}"
+
+      if [ "$status" = "green" ]; then
+        log "OpenSearch restore complete and cluster is healthy."
+        return
+      fi
+
+      sleep 30
+      retries=$((retries - 1))
+    done
+
+    log "ERROR: Timeout waiting for OpenSearch cluster to become healthy after restore."
+    exit 1
+    ;;
+  harbor)
+    log "Starting Harbor restore..."
+
+    "$CK8S_CMD" harbor-restore
+
+    log "Harbor restore successfully completed."
+    ;;
+  rclone)
+    log "Starting Rclone restore..."
+
+    # Check if rclone-restore is enabled before continuing
+    log "Checking for rclone-restore cronjobs"
+    local cronjob_list
+    cronjob_list=$("$CK8S_CMD" ops kubectl sc -n rclone get cronjobs -lapp.kubernetes.io/instance=rclone-restore -oname)
+
+    if [ -z "$cronjob_list" ]; then
+      log "ERROR: No rclone-restore cronjobs found."
+      log "Please ensure rclone restore is enabled in your configuration and has been applied."
+      exit 1
+    fi
+    local total_jobs
+    total_jobs=$(echo "$cronjob_list" | wc -w)
+    log "Found ${total_jobs} cronjobs."
+
+    # Create jobs from cronjobs
+    for cronjob in $cronjob_list; do
+      "$CK8S_CMD" ops kubectl sc -n rclone create job --from "${cronjob}" "${cronjob/#cronjob.batch\//}"
+      local job_name=${cronjob/#cronjob.batch\//}
+      log "Creating job '${job_name}' from '${cronjob}' cronjob..."
```

Comment:
Question: Would it make sense to have the log before running the command

```suggestion
      local job_name=${cronjob/#cronjob.batch\//}
      log "Creating job '${job_name}' from '${cronjob}' cronjob..."
      "$CK8S_CMD" ops kubectl sc -n rclone create job --from "${cronjob}" "${job_name}"
```

Alternatively change the wording to `Created job...`

Reply:
I made it easy and just changed the wording to created job. Since kubectl create job already spits out that it creates/created the job I am not sure if there's even a log necessary here. But I think it looks nice for the script itself to at least say that it created the job after the kubectl command finishes for logging/debugging purposes.